core/src/main/java/org/elasticsearch/action/ingest/IngestDisabledActionFilter.java

```diff
@@ -52,7 +52,7 @@ public int order() {
     }
 
     private static void failRequest(String pipelineId) {
-        throw new IllegalArgumentException("ingest plugin is disabled, cannot execute pipeline with id [" + pipelineId + "]");
```

Comment:
Yes, redirecting is much better than failing. It has been discussed (it is a small not on the meta issue) and we improve this after the move ingest into core refactoring has been completed.